cdap-data-fabric/src/main/java/co/cask/cdap/store/DefaultNamespaceStore.java

```diff
@@ -124,25 +135,22 @@ public void apply(NamespaceMDS mds) throws Exception {
           }
         }
       }, apps.get());
+    // refresh the namespacemeta in the cache
+    namespaceMetaCache.refresh(metadata.getNamespaceId());
```

Comment:
We can but I thought about that but decided to refresh  since the chance of having a get after update is high and then get will take less time. Either is fine I guess.